src/modules/user/auth.service.ts

```diff
@@ -0,0 +1,234 @@
+import {
+  ConflictException,
+  ForbiddenException,
+  HttpException,
+  HttpStatus,
+  Injectable,
+  NotFoundException,
+  UnauthorizedException,
+  UnprocessableEntityException,
+} from '@nestjs/common';
+import { randomBytes, scrypt as _scrypt } from 'crypto';
+import { promisify } from 'util';
+import { ConfigService } from '@nestjs/config';
+
+import { UserService } from './user.service';
+import { EmailService } from '../email/email.service';
+import { AuthHelper } from './auth.helper';
+import { UserDal } from './user.dal';
+import { CreateUserDto } from './dto/create-user.dto';
+import { errorMessage } from '../../constants/errorMessages';
+import { throwError } from '../.././modules/helpers/throwError';
+import { ConfirmEmailDto } from './dto/confirm-email.dto';
+import { UpdateUserDto } from './dto/update-user.dto';
+import { Entities } from 'src/constants/entities';
+import { Properties } from 'src/constants/properties';
+import { controller } from 'src/constants/controller';
+import { controller_path } from 'src/constants/controllerPath';
+import { ResetPasswordDto } from './dto/reset-password.dto';
+import { LogInUserDto } from './dto/login-user.dto';
+
+const scrypt = promisify(_scrypt);
+
+@Injectable()
+export class AuthService {
+  constructor(
+    private usersService: UserService,
+    private usersDal: UserDal,
+    private emailService: EmailService,
+    private authHelper: AuthHelper,
+  ) {}
+
+  async createUser(createUserDto: CreateUserDto) {
+    try {
+      console.log(process.env.TOKEN_SECRET_KEY);
+      const { email, password } = createUserDto;
+
+      const user = await this.usersDal.findByEmail(email);
+
+      if (user) {
+        throw new UnprocessableEntityException(
+          errorMessage.EMAIL_IN_USE(email),
+        );
+      }
+      const salt = randomBytes(8).toString('hex');
+      const hash = (await scrypt(password, salt, 32)) as Buffer;
+      const result = salt + '.' + hash.toString('hex');
+
+      const registerUser: CreateUserDto = {
+        ...createUserDto,
+        password: result,
+      };
+      const registeredUser = await this.usersDal.create(registerUser);
+
+      const newUser =
+        await this.authHelper.updateConfirmationTokenAndReturnNewUser(
+          registeredUser,
+        );
+
+      const confirmationToken =
+        await this.authHelper.sendConfirmationEmail(newUser);
+
+      return confirmationToken;
+    } catch (error) {
+      throwError(error.status, error.message);
```

Comment:
@gerimujo496 Redundant, we just rethrow the error we don't handle it. This could be done in the controller directly